src/App.js

```diff
@@ -6,26 +6,31 @@ function App() {
   const [error, setError] = useState(null);
   const [isLoaded, setIsLoaded] = useState(false);
   const [city, setCity] = useState("New York City")
-  const [results, setResults] = useState(null);
+  const [results, setResults] = useState([]);
+  const [buttonClick, setButtonClick] = useState(false);
+
+  const APIKEY = process.env.REACT_APP_APIKEY;
+
+  console.log(APIKEY);
 
   useEffect(() => {
-    fetch("https://api.openweathermap.org/data/2.5/weather?q=" + city + "&units=metric" + "&appid=" + process.env.REACT_APP_APIKEY)
+    fetch("https://api.openweathermap.org/data/2.5/weather?q=" + city + "&units=metric" + "&appid=" + APIKEY)
```

Comment:
Code improvement suggestion: 
1. use template literals (a.k.a backticks) for string concatenation
2. the variable `APIKEY` is redundant because you are using it to pass value of another variable "`process.env.REACT_APP_APIKEY`", so please **remove** that and just use `process.env.REACT_APP_APIKEY`